packages/manager/apps/hpc-vmware-vsphere/src/pages/dashboard/index.tsx

```diff
@@ -47,15 +93,53 @@ export default function DashboardPage() {
       .reverse()
       .find((tab) => location.pathname.startsWith(tab.to));
     if (activeTab) {
-      setActivePanel(activeTab.to);
+      if (activeTab.isRedirectLegacy) {
+        const loadLegacyApp = async () => {
+          const legacyAppUrl = (await shellNav.getURL(
+            'dedicated',
+            `#/dedicated_cloud/${serviceName ?? ''}/${activeTab.to}`,
+            {},
+          )) as string;
+          window.location.href = legacyAppUrl;
+        };
```

Comment:
You can use directly:  packages/manager/apps/hycu/src/hooks/shell/useNavigationGetUrl.ts

```tsx 
const legacyAppUrl = useNavigationGetUrl(['dedicated',
            `#/dedicated_cloud/${serviceName ?? ''}/${activeTab.to}`,
            {}, {enabled: () => activeTab.isRedirectLegacy}];
```
            
And this allow you to remove `const shellNav = useContext(ShellContext).shell.navigation;`